test/issues/general/test_9854.test

```diff
@@ -0,0 +1,337 @@
+# group: [general]
+
+statement ok
+create table tbl (a varchar, b varchar);
+
+statement ok
+insert into tbl values ('ö', '>>>>>ö<<<<<'), ('o', '>>>>>o<<<<<'), ('p', '>>>>>p<<<<<');
+
+######### !!! test concat()
+
+query I
+select a from tbl order by a;
+----
+o
+p
+ö
+
+query I
+select a from tbl order by a collate noaccent;
+----
+ö
+o
+p
+
+query I
+select a from tbl order by concat(a collate noaccent, a);
+----
+ö
+o
+p
+
+query I
+select a from tbl order by concat(a, a collate noaccent);
+----
+ö
+o
+p
+
+query I
+select a from tbl order by concat(a, a collate noaccent);
+----
+ö
+o
+p
+
+query I
+select a from tbl order by concat(concat(a, a collate noaccent), a);
+----
+ö
+o
+p
+
+query I
+select a from tbl order by concat(a collate noaccent, a collate noaccent);
+----
+ö
+o
+p
+
+statement error
+select a from tbl order by concat(a collate noaccent, a collate nocase);
+----
+
+query I
+select a from tbl order by a collate noaccent;
+----
+ö
+o
+p
+
+######### !!! test lower() and upper()
+
+query I
+select a from tbl order by lower(a);
+----
+o
+p
+ö
+
+query I
+select a from tbl order by lower(a collate noaccent);
+----
+ö
+o
+p
+
+query I
+select a from tbl order by upper(a);
+----
+o
+p
+ö
+
+query I
+select a from tbl order by upper(a collate noaccent);
+----
+ö
+o
+p
+
+######### !!! test trim(), ltrim() and rtrim()
+
+query I
+select a from tbl order by trim(b);
+----
+o
+p
+ö
+
+query I
+select a from tbl order by trim(b, '><');
+----
+o
+p
+ö
+
+query I
+select a from tbl order by trim(b collate noaccent);
+----
+ö
+o
+p
+
+query I
+select a from tbl order by trim(b collate noaccent, '><');
+----
+ö
+o
+p
+
+query I
+select a from tbl order by trim(b, '><' collate noaccent);
+----
+ö
+o
+p
+
+statement error
+select a from tbl order by trim(b collate nocase, '><' collate noaccent);
+----
+
+query I
+select a from tbl order by ltrim(b);
+----
+o
+p
+ö
+
+query I
+select a from tbl order by ltrim(b, '><');
+----
+o
+p
+ö
+
+query I
+select a from tbl order by ltrim(b collate noaccent);
+----
+ö
+o
+p
+
+query I
+select a from tbl order by ltrim(b collate noaccent, '><');
+----
+ö
+o
+p
+
+query I
+select a from tbl order by ltrim(b, '><' collate noaccent);
+----
+ö
+o
+p
+
+statement error
+select a from tbl order by ltrim(b collate nocase, '><' collate noaccent);
+----
+
+query I
+select a from tbl order by rtrim(b);
+----
+o
+p
+ö
+
+query I
+select a from tbl order by rtrim(b, '><');
+----
+o
+p
+ö
+
+query I
+select a from tbl order by rtrim(b collate noaccent);
+----
+ö
+o
+p
+
+query I
+select a from tbl order by rtrim(b collate noaccent, '><');
+----
+ö
+o
+p
+
+query I
+select a from tbl order by rtrim(b, '><' collate noaccent);
+----
+ö
+o
+p
+
+statement error
+select a from tbl order by rtrim(b collate nocase, '><' collate noaccent);
+----
+
+query I
+select trim(b, '>o') from tbl;
+----
+ö<<<<<
+<<<<<
+p<<<<<
+
+query I
+select trim(b, '>o' collate noaccent) from tbl;
+----
+<<<<<
+<<<<<
+p<<<<<
```

Comment:
I don't think this result is correct. In the sample query, the collation has been applied only into the pattern argument (2nd argument), but the first one is not collated. For that reason, I believe the correct output should be:

```
D select trim(b, '>o' collate noaccent) from tbl;
┌─────────────────────────────────────┐
│ main.trim(b, '>o' COLLATE noaccent) │
│               varchar               │
├─────────────────────────────────────┤
│ ö<<<<<                              │
│ <<<<<                               │
│ p<<<<<                              │
└─────────────────────────────────────┘
```
But, if you apply the collation to the first argument like bellow, the output is proper:
```

D select trim(b collate noaccent, '>o') from tbl;
┌─────────────────────────────────────┐
│ main.trim(b COLLATE noaccent, '>o') │
│               varchar               │
├─────────────────────────────────────┤
│ <<<<<                               │
│ <<<<<                               │
│ p<<<<<                              │
└─────────────────────────────────────┘
```